### ISSUE TITLE
Bad table config may result in Zombie tables

### ISSUE THREAD

**Issue by mayankshriv** [apache/pinot#8500]
Issue faced by user:

```
hey folks, does anyone have experience with “zombie” tables?  I attempted to add a tableSpec but was missing a field in the
config.  This resulted in a null pointer exception when Pinot processed the config.  As a result, the table doesn’t exist in the 
UI or by querying the list API, but I can’t create a table with the same name due to receiving an error that “table already 
exists”.  Presumably the spec reached ZK or something, but no where else? What can be done to remove the zombie table?
```

**Comment by mayankshriv** [comment]
@timsants could you please take a look? Some how I am unable to assign this to you.

**Comment by timsants** [comment]
Investigation in progress

**Comment by timsants** [comment]
When a realtime table is being created, the table config is first uploaded to the ZK property store. After that, we try to create the realtime consumer. But as pointed out, if there is an invalid config for the consumer, an exception is thrown. The next time you try to create the table (with correct configs), it checks the ZK property store for the existence of the table config and throws an exception saying that the table already exists (although the table creation process did not finish).

**Why does the table not show up with GET `/tables`?**
It checks for the tables existence in the ideal state [resources](https://github.com/apache/pinot/blob/master/pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java#L604). Because the realtime table creation process never completed, the realtime table was not added to the ZK ideal state.

**Why did deleting the table not fix the issue?**
Table deletion should have actually fixed the issue because it deletes the config from the property store (verified that this was the case). The return message from the operation was misleading because it only returns a success message if the table was originally in the ideal state.

**Proposed solution**
The table existence check for table creation is inconsistent with the other table operations (GET and DELETE). For table creation, we should check the table existence in the ideal state instead of checking for the table config in the property store. If the table config already existed in the property store, it should be ok to overwrite since it was never fully created.

**Comment by Jackie-Jiang** [comment]
Checking table config existence is intentional to prevent race condition of re-creating a recently not fully deleted table (see #8103 for details)
The correct fix here should be:
- If adding a table fails, delete the table config and all entries already persisted
- Enhance the exception message to inform the user to try delete the table which can delete the remaining entries

**Comment by npawar** [comment]
@timsants there was a report that this worked fine in 0.9.3 (i.e. table creation failed cleanly) and in 0.10.0 this fails with zombie states. Any idea why that is?

**Comment by timsants** [comment]
@npawar, this was a change that was merged in Feb [#8103](https://github.com/apache/pinot/pull/8103/files). Before that, realtime tables didn't have this check. Previously, the table config was probably being overwritten but I don't think the other ZK metadata was being cleaned up correctly.

**Comment by mayankshriv** [comment]
Fixed by #8509

**Comment by timsants** [comment]
Fix has been merged. We now do a full ZK metadata cleanup in case there is any failure during the table creation process.
